botorch/utils/gp_sampling.py

```diff
@@ -197,44 +199,59 @@ def _get_weights(
         return weights
 
     def forward(self, X: Tensor) -> Tensor:
-        """
-        Get fourier basis features for the provided inputs.
-        Note that if `sample_shape` has been passed, then the rightmost
-        subset of the batch shape of the input should be `sample_shape`.
+        """Get fourier basis features for the provided inputs.
+
+        Note that the right-most subset of the batch shape of `X`
+        should be `sample_shape x kernel_batch_shape` if using either
+        the `sample_shape` argument or a batched kernel.
```

Comment:
What if you're using the `sample_shape` argument but not a batched kernel or vice versa? Is the "right-most subset" always 2 elements or can it be 1? Some examples of what `X.shape` should look like would clarify.